src/Components/Components/src/Routing/NavLink.cs

```diff
@@ -49,29 +53,38 @@ public class NavLink : IComponent, IDisposable
         [Inject] private IUriHelper UriHelper { get; set; }
 
         /// <inheritdoc />
-        public void Configure(RenderHandle renderHandle)
+        protected override void OnInit()
         {
-            _renderHandle = renderHandle;
-
             // We'll consider re-rendering on each location change
             UriHelper.OnLocationChanged += OnLocationChanged;
         }
 
         /// <inheritdoc />
-        public Task SetParametersAsync(ParameterCollection parameters)
+        public override Task SetParametersAsync(ParameterCollection parameters)
         {
-            // Capture the parameters we want to do special things with, plus all as a dictionary
-            parameters.TryGetValue(RenderTreeBuilder.ChildContent, out _childContent);
-            parameters.TryGetValue("class", out _cssClass);
-            parameters.TryGetValue("href", out string href);
-            ActiveClass = parameters.GetValueOrDefault(nameof(ActiveClass), DefaultActiveClass);
-            Match = parameters.GetValueOrDefault(nameof(Match), NavLinkMatch.Prefix);
-            _allAttributes = parameters.ToDictionary();
-
-            // Update computed state and render
+            // Run the default parameter setting logic to populate our properties.
+            base.SetParametersAsync(parameters);
```

Comment:
Why no await here or return the resulting task?

Reply:
Or to make this simpler, why not override `OnParametersSetAsync` instead of `SetParametersAsync`? The logic here doesn't use the `ParameterCollection` directly. Also then you wouldn't have to call any `base` method.

Reply:
overriding `OnParametersSet` now